### DISCUSSION TITLE
How to add custom language support (lsp)

### DISCUSSION THREAD

**Discussion by un4gt** [eclipse-theia/theia#14231] [Q&A] [answered]
Before starting this discussion, I checked some previous discussion related to lsp . I don't want to import vscode plugins to achieve this , just get through with theia extension, but I can't find any content about language support on theia document . Can anyone give me some ideas?

**Comment by msujew** [comment, marked as answer]
Hey @un4gt,

the recommended way is to use VS Code plugins. In theory, you could create all the bindings to the monaco editors and all the other widgets yourself, but it's not recommended (and likely very difficult).

**Reply by un4gt** [reply]
Thank you @msujew，  i will try to use `VS Code plugins` to solve this .

**Reply by johnholliday** [reply]
As I understand it, the VSIX approach is limited with regard to full Theia IDE integration.  Is there a complete example of integrating a DSL as a Theia plugin?

**Reply by msujew** [reply]
I don't think there's a good example around - if you need to combine a vscode plugin and a Theia extension, there are ways to do this by passing data around via commands.
